Updating the SPM guide for iOS and macOS (#13459)
flutter/website: sites/docs/src/content/packages-and-plugins/swift-package-manager/for-app-developers.md

description: How to use Swift Package Manager for native iOS or macOS dependencies
---

:::note
As of the 3.44 release, Flutter uses [Swift Package Manager][]
to manage iOS and macOS native dependencies.
Flutter continues to support CocoaPods in maintenance mode,
however, the CocoaPods registry permanently becomes
[read-only on December 2, 2026][cocoapods].
:::

[cocoapods]: https://blog.cocoapods.org/CocoaPods-Specs-Repo/
[Swift Package Manager]: https://www.swift.org/documentation/package-manager/

## How to turn on Swift Package Manager

> As of the 3.44 release, Flutter's Swift Package Manager (SPM)

Reviewer: The official abbreviation for Swift Package Manager is "SwiftPM"
```suggestion
As of the 3.44 release, Flutter's Swift Package Manager (SwiftPM)
```

support is on by default.

> Upgrading Flutter and running your app automatically adds SPM integration.

Reviewer: ```suggestion
Upgrading Flutter and running your app automatically adds SwiftPM integration.
```

This makes your project download the Swift packages that
your Flutter plugins depend on.
To use an older Flutter version,

> you must [remove Swift Package Manager integration][removeSPM]

Reviewer: ```suggestion
you may need to [remove Swift Package Manager integration][removeSPM]
```

SwiftPM support has been available as an experimental feature for a long time.

from the app.

Note that Flutter falls back to CocoaPods for dependencies that don't
yet support Swift Package Manager.

> [Optional] To check if your project is using SPM:

Reviewer: ```suggestion
[Optional] To check if your project is using SwiftPM:
```

1. In Xcode, run the app.
1. Ensure that  **Run Prepare Flutter Framework Script** runs as a pre-action
   and that `FlutterGeneratedPluginSwiftPackage` is a target dependency.

   <DashImage image="development/packages-and-plugins/swift-package-manager/flutter-pre-action-build-log.png" caption="Ensure **Run Prepare Flutter Framework Script** runs as a pre-action" />

If automatic migration works for you, that's it!

> You are done with this page.

Reviewer: Maybe add something here about if you had previously disabled SwiftPM? Like

"If you have previously disabled SwiftPM, you may need to enable it with `flutter config --enable-swift-package-manager`"

> ## How to turn off Swift Package Manager

Reviewer: Maybe we move this closer to the bottom? Since we don't want them to do it?

In general, don't do this. Remember that

> the CocoaPods registry becomes read-only on December 2, 2026.

Reviewer: Maybe something here about how this will be disallowed in a future version of Flutter?

Disabling Swift Package Manager causes Flutter to use CocoaPods for all

> dependencies. However, SPM remains integrated with your project.

Reviewer: ```suggestion
dependencies. However, SwiftPM remains integrated with your project.
```

To remove Swift Package Manager integration completely from your project,
follow the [How to remove Swift Package Manager integration][removeSPM]
instructions.

> ### Turn off SPM for a single project

Reviewer: ```suggestion
### Turn off SwiftPM for a single project
```

In the project's `pubspec.yaml` file, under the `flutter` section,
set `enable-swift-package-manager` to `false` in the `config` subsection.

```yaml title="pubspec.yaml"
# The following section is specific to Flutter packages.
flutter:
  config:
    enable-swift-package-manager: false
```

This turns off Swift Package Manager for all contributors to this project.

> ### Turn off SPM globally for all projects

Reviewer: ```suggestion
### Turn off SwiftPM globally for all projects
```

Run the following command:

```sh
flutter config --no-enable-swift-package-manager
```

This turns off Swift Package Manager for the current user.

If a project is incompatible with Swift Package Manager,
all contributors need to run this command.

[removeSPM]: #how-to-remove-swift-package-manager-integration

> ## How to add Swift Package Manager integration

Reviewer: ```suggestion
## How to add Swift Package Manager integration manually
```

When you upgrade to Flutter 3.44 or later and run
your app, SPM integration is automatically added. You only need
these instructions if problems occurred and you need to manually
add SPM integration to your project.

Most developers do not need to do this.

If you experienced a problem automatically migrating your

> project to SPM, please [file an issue][].

Reviewer: ```suggestion
project to SwiftPM, please [file an issue][].
```

Include the error message and, if possible,
include a copy of the following files in your issue:

* `ios/Runner.xcodeproj/project.pbxproj`
* `ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme`
   (or the xcsheme for the flavor used)

> ### Add to a Flutter app

Reviewer: This section does not seem to provide any additional needed content.

```suggestion
```

> Use the following instructions to manually migrate your app.

Reviewer: ```suggestion
```

> ### Add to a Flutter app manually

Reviewer: ```suggestion
```

### Add to an existing app (add-to-app)

> To use SPM, consult one of the following pages, as appropriate:

Reviewer: ```suggestion
To use SwiftPM, consult one of the following pages, as appropriate:
```

You can add Swift Package Manager integration to these custom Xcode targets.

Follow the steps in
[How to add Swift Package Manager integration][manualIntegration].

In [Step 1][] for list item 6,
use your custom target instead of the `Flutter` target.

In [Step 2][] for list item 6,
use your custom target instead of the `Flutter` target.

[Xcode targets]: {{site.apple-dev}}/documentation/xcode/configuring-a-new-target-in-your-project
[manualIntegration]: #how-to-add-swift-package-manager-integration
[Step 1]: #step-1-add-fluttergeneratedpluginswiftpackage-package-dependency
[Step 2]: #step-2-add-run-prepare-flutter-framework-script-pre-action

## How to remove Swift Package Manager integration

> When your app is modified to support SPM,

Reviewer: ```suggestion
When your app is modified to support SwiftPM,
```
